src/type.cpp

```diff
@@ -158,6 +158,34 @@ std::string StructType::id() const noexcept {
   return id_;
 }
 
+std::size_t StructType::offset(const std::string& id) const {
+  std::size_t offset = 0;
+  for (auto i = std::size_t{0}, e = fields_.size(); i < e; ++i) {
+    const auto& field = fields_.at(i);
+    if (field->id == id) {
+      return offset;
+    }
+
+    offset += field->type->size();
+  }
+
+  throw "member not found in struct!";
```

Comment:
Consider throwing an `exception` instead of a `string`. The basic one is `std::runtime_error`.